### PR TITLE
style: Move font-variant-east-asian outside of mako

### LINKED ISSUE #19216: Move font-variant-east-asian outside of mako.
I'll take this one as a sub-issue of #19015

### PR DESCRIPTION
**Pull request by CYBAI**
This is a sub-PR of #19015 
r? emilio 

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #19216 
- [x] These changes do not require tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19220)

### CODE REVIEW

**Comment by nox** [comment]
There is a typo in the commit title: "font-vairant-east-asian".

**Comment by CYBAI** [comment]
@nox Oh, thanks! Let me update it!

File: components/style/values/specified/font.rs [hunk 1]
```diff
@@ -851,6 +851,318 @@ impl Parse for FontVariantAlternates {
...
+        const RUBY = 0x100;
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl VariantEastAsian {
+    /// Obtain a specified value from a Gecko keyword value
+    ///
+    /// Intended for use with presentation attributes, not style structs
+    pub fn from_gecko_keyword(kw: u16) -> Self {
```
Comment (emilio) [inline components/style/values/specified/font.rs]: Yeah, this is not great, the right way to do it is having the same bytes as the gecko keyword, and using `VariantEastAsian::from_bits_truncate` and `VariantEastAsiant::bits` to go from / to `u16`.

The cool part is, they match already I believe!

But we need to assert it so it doesn't break, so I'd add a `pub fn assert_variant_east_asian_matches()` similar to:

  https://github.com/servo/servo/blob/fe4139b779b3af749ec1426ddf4e1393c7b85442/components/style/invalidation/element/restyle_hints.rs#L238

And then make this function just:

```rust
Self::from_bits_truncate(kw)
```

And the `to_gecko_keyword` just:

```rust
self.bits()
```

**Review by emilio — APPROVED** [review]
Looks good! But let's make the patch a bit smaller using the fact that the representations match.

File: components/style/values/specified/font.rs (hunk 1 above)
Comment (CYBAI) [reply components/style/values/specified/font.rs]: Updated!

**Comment by emilio** [comment]
@bors-servo r+

Neat!

**Comment by jdm** [comment]
@bors-servo retry
* failure to build random crate

**Pull request merged**